files/secure_media_views.py

```
@@ -353,6 +359,55 @@ def _is_valid_file_path(self, file_path: str) -> bool:
+        # Check if the requested path exactly matches any of the media's thumbnail paths
+        if file_path in thumbnail_paths:
+            return True
+
+        # Also check for encoded GIF paths
+        if file_path.startswith('encoded/') and file_path.lower().endswith('.gif'):
+            # For encoded GIFs, verify via the Encoding model
+            # The path format is: encoded/{profile_id}/{username}/{filename}
+            from .models import Encoding
+            parts = file_path.split('/')
+            if len(parts) >= 4:
+                filename = parts[3]
```

Ruff (0.14.13)

Local variable `filename` is assigned to but never used

Remove assignment to unused variable `filename`

(F841)